EIPS/eip-holdable-token.md

```diff
@@ -0,0 +1,293 @@
+
+## Motivation
+
+A hold has to be used in different scenarios where a immediate transfer between accounts is not possible or has to be guaranteed beforehand:
+
+1. A regulated token may not allow to do a token transfer between accounts without verifying first, that it follows all the regulations. In this case a clearable transfer has to be used. During the clearing process a hold is created to ensure, that the transfer is successful after all checks have passed. If the transfer violates any of the regulations, it is cleared and not further processed. 
+
+1. In certain business situations a payment has to be guaranteed before its services can be used. For example: When checking in a hotel, the hotel will put a hold on the guest's account to ensure that enough balance is available to pay for the room before handing over the keys.
+
+1. In other occasions a payment has to be guaranteed without knowing the exact amount beforehand. To stay with the hotel example: The hotel can put a hold on the guest's account as a guarantee for any possible extras, like room service. When the guest checks out the hold is partially executed and the remaining amount is available again on the guest's account.
+
+The ERC-20 `approve` function provides some of the necessary functionality for the use cases above. The main difference to holds, is that `approve` does not ensure a payment, as the approved money is not blocked and can be transferred at any moment.
+
+## Specification
+
+```solidity
+interface IHoldable /* is ERC-20 */ {
+    enum HoldStatusCode {
+        Nonexistent,
+        Ordered,
+        Executed,
+        ReleasedByNotary,
+        ReleasedByPayee,
+        ReleasedOnExpiration
+    }
+
+    function hold(string calldata operationId, address to, address notary, uint256 value, bool expires, uint256 timeToExpiration) external returns (bool); 
+    function holdFrom(string calldata operationId, address from, address to, address notary, uint256 value, bool expires, uint256 timeToExpiration) external returns (bool);
+    function releaseHold(address holdIssuer, string calldata operationId) external returns (bool);
+    function executeHold(address holdIssuer, string calldata operationId, uint256 value) external returns (bool);
+    function renewHold(string calldata operationId, uint256 timeToExpiration) external returns (bool);
+    function retrieveHoldData(address holdIssuer, string calldata operationId) external view returns (address from, address to, address notary, uint256 value, bool expires, uint256 expiration, HoldStatusCode status);
+
+    function balanceOnHold(address account) external view returns (uint256);
+    function netBalanceOf(address account) public view returns (uint256);
+    function totalSupplyOnHold() external view returns (uint256);
+
+    function authorizeHoldOperator(address operator) external returns (bool);
```

Comment:
If we changed this names to be compliant with erc777, why dont make it explicit at the README?

Reply:
I've added it to rationale